server/middleware/populateReport.test.ts

```diff
@@ -22,27 +23,43 @@ describe('report-loading middleware', () => {
     nock.cleanAll()
   })
 
-  it('should call next request handler if report can be loaded', async () => {
-    fakeApi.get(`/incident-reports/${basicReport.id}`).reply(200, basicReport)
+  it('calls next request handler if report can be loaded', async () => {
+    fakeApi.get(`/incident-reports/${report.id}/with-details`).reply(200, report)
 
-    const req = { params: { id: basicReport.id } } as unknown as Request
+    const req = { params: { id: report.id } } as unknown as Request
     const res = { locals: { apis: { incidentReportingApi: new IncidentReportingApi('token') } } } as Response
     const next: NextFunction = jest.fn()
     await populateReport()(req, res, next)
 
     expect(next).toHaveBeenCalledWith()
-    expect(res.locals.incident.reportReference).toEqual(basicReport.reportReference)
+    expect(res.locals.report.reportReference).toEqual(report.reportReference)
+    expect(res.locals.reportConfig.incidentType).toEqual(report.type)
   })
 
-  it('should forward error if report cannot be loaded', async () => {
-    fakeApi.get(`/incident-reports/${basicReport.id}`).reply(404)
+  it('forwards error if report cannot be loaded', async () => {
+    fakeApi.get(`/incident-reports/${report.id}/with-details`).reply(404)
 
-    const req = { params: { id: basicReport.id } } as unknown as Request
+    const req = { params: { id: report.id } } as unknown as Request
     const res = { locals: { apis: { incidentReportingApi: new IncidentReportingApi('token') } } } as Response
     const next: NextFunction = jest.fn()
     await populateReport()(req, res, next)
 
     expect(next).toHaveBeenCalledWith(expect.objectContaining({ message: 'Not Found', status: 404 }))
-    expect(res.locals.incident).toBeUndefined()
+    expect(res.locals.report).toBeUndefined()
+    expect(res.locals.reportConfig).toBeUndefined()
+  })
+
+  it('forwards error if report config cannot be loaded', async () => {
+    report.type = 'UNKNOWN' as Type
+    fakeApi.get(`/incident-reports/${report.id}/with-details`).reply(200, report)
+
+    const req = { params: { id: report.id } } as unknown as Request
+    const res = { locals: { apis: { incidentReportingApi: new IncidentReportingApi('token') } } } as Response
+    const next: NextFunction = jest.fn()
+    await populateReport()(req, res, next)
+
+    expect(next).toHaveBeenCalledWith(expect.objectContaining({ code: 'MODULE_NOT_FOUND' }))
```

Comment:
is the next function called with an `Error` instance? dunno how stable/universal "MODULE_NOT_FOUND" is… so maybe `expect.any(Error)` is good enough? does `expect.objectContaining({ code: 'MODULE_NOT_FOUND' })` feel a little brittle to you?

Reply:
Not different from the similar assertion in the other test tbh. `any(Error)` risks to catch possibly other things so I'd rather have the more specific assertion and if the error ever change we'll cross that bridge. IMHO of course.